plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1756,405 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part = 1 OR x = 'a' ", "VALUES 5");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part = 1 AND  x = 'a' ", "VALUES 1");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part IS NOT NULL", "VALUES 8");
+            assertQuery(session, "SELECT x, count(*) AS COUNT FROM " + table.getName() + " WHERE part > 2 GROUP BY x ", "VALUES ('a', 2), ('b', 2)");
+            assertQueryFails(session, "SELECT count(*) FROM " + table.getName() + " WHERE x= 'a'", "Filter required on .*" + table.getName() + " for at least one partition column:.*");
+        }
+    }
+
+    @Test
+    public void testRequiredPartitionFilterOnJoin()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+
+        try (TestTable table1 = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_left_",
+                "(x varchar, part varchar)",
+                ImmutableList.of("('a', 'part_a')"));
+                    TestTable table2 = new TestTable(
+                            new TrinoSqlExecutor(getQueryRunner(), session),
+                           "test_partition_right_",
+                            "(x varchar, part varchar) WITH (partitioned_by = ARRAY['part'])",
+                            ImmutableList.of("('a', 'part_a')"))) {
+            assertQueryFails(session, "SELECT a.x, b.x from %s a JOIN %s b on (a.x = b.x) where a.x = 'a'".formatted(table1.getName(), table2.getName()), "Filter required on .*" + table2.getName() + " for at least one partition column:.*");
+            assertQuery(session, "SELECT a.x, b.x from %s a JOIN %s b on (a.part = b.part) where a.part = 'part_a'".formatted(table1.getName(), table2.getName()), "VALUES ('a', 'a')");
+        }
+    }
+
+    @Test
+    public void testRequiredPartitionFilterOnJoinBothTablePartitioned()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+
+        try (TestTable table1 = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_inferred_left_",
+                "(x varchar, part varchar) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("('a', 'part_a')"));
+                    TestTable table2 = new TestTable(
+                            new TrinoSqlExecutor(getQueryRunner(), session),
+                            "test_partition_inferred_right_",
+                            "(x varchar, part varchar) WITH (partitioned_by = ARRAY['part'])",
+                            ImmutableList.of("('a', 'part_a')"))) {
+            assertQueryFails(session, "SELECT a.x, b.x from %s a JOIN %s b on (a.x = b.x) where a.x = 'a'".formatted(table1.getName(), table2.getName()), "Filter required on .*" + table1.getName() + " for at least one partition column:.*");
```

Comment:
This line is too long. Please wrap per arguments. Please make sure other places as well.